README.md

```diff
@@ -686,6 +689,18 @@ Some more details on how the navigation API handles scrolling with `"traverse"`
 
 - By default, any navigations which are intercepted with `navigateEvent.intercept()` will _ignore_ the value of `history.scrollRestoration` from the classic history API. This allows developers to use `history.scrollRestoration` for controlling cross-document scroll restoration, while using the more-granular option to `intercept()` to control individual same-document navigations.
 
+#### Manual commit
+The default behavior of immediately "committing" (i.e., updating `location.href` and `navigation.currentEntry`) works well for most situations, but some developers may find they do not want to immediately update the url, and may want to retain the option of aborting the navigation without needing to rollback a url update or cancel-and-restart. This behavior can be customized using `intercept()`'s `commit` option:
+
+- `e.intercept({ handler, commit: "immediate" })`: the default behavior, immediately commit the navigation and update `location.href` and `navigation.currentEntry`.
+- `e.intercept({ handler, commit: "manual" })`: start the navigation (e.g., show a loading spinner if the UI has one), but do not immediately commit.
+
+When manual commit is used, the navigation will commit (and a `committed` promise will resolve if present) when `e.commit()` is called. If any handler(s) passed to `intercept()` fulfill, and `e.commit()` has not yet been called, we will fallback to committing just before any `finish` promise resolves and `navigatesuccess` is fired.
+
+If a handler passed to `intercept()` rejects before `e.commit()` is called, then the navigation will be treated as canceled (both `committed` and `finished` promises will reject, and no url update will occur). If a handler passed to `intercept()` rejects after `e.commit()` is called, the behavior will match a rejected promise in immediate commit mode (i.e., the `committed` promise will fulfill, the `finished` promise will reject, and the url will update).
+
+Because manual commit can be used to cancel the navigation before the url updates, it is only available when `e.cancelable` is true.
```

Comment:
```suggestion
Because manual commit can be used to cancel the navigation before the URL updates, it is only available when `e.cancelable` is true. See [above](#restrictions-on-firing-canceling-and-responding) for details on when `e.cancelable` is set to false, and thus manual commit is not available.
```

Also state what will happen if you try to use manual commit when `e.cancelable` is false. An exception, I assume.

Reply:
I haven't finalized that. My draft does a console warning (on the logic that it's better to invalidate the commit option, rather than invalidate the entire intercept). Is it ok to leave that ambiguous in the explainer (since we don't really enumerate every possible exception in other parts of the API).